controllers/registry.go

```diff
@@ -270,30 +284,54 @@ func (r *VeleroReconciler) buildRegistryContainer(bsl *velerov1.BackupStorageLoc
 		},
 	}
 
-	return containers
+	return containers, nil
 }
 
-func (r *VeleroReconciler) getRegistryEnvVars(bsl *velerov1.BackupStorageLocation) []corev1.EnvVar {
+func (r *VeleroReconciler) getRegistryEnvVars(bsl *velerov1.BackupStorageLocation) ([]corev1.EnvVar, error) {
 	envVar := []corev1.EnvVar{}
 	provider := bsl.Spec.Provider
+	var err error
 	switch provider {
 	case AWSProvider:
-		envVar = r.getAWSRegistryEnvVars(bsl, cloudProviderEnvVarMap[AWSProvider])
+		envVar, err = r.getAWSRegistryEnvVars(bsl, cloudProviderEnvVarMap[AWSProvider])
 
 	case AzureProvider:
 		envVar = r.getAzureRegistryEnvVars(bsl, cloudProviderEnvVarMap[AzureProvider])
 
 	case GCPProvider:
 		envVar = r.getGCPRegistryEnvVars(bsl, cloudProviderEnvVarMap[GCPProvider])
 	}
-	return envVar
+	if err != nil {
+		return nil, err
+	}
+	return envVar, nil
 }
 
-func (r *VeleroReconciler) getAWSRegistryEnvVars(bsl *velerov1.BackupStorageLocation, awsEnvVars []corev1.EnvVar) []corev1.EnvVar {
+func (r *VeleroReconciler) getAWSRegistryEnvVars(bsl *velerov1.BackupStorageLocation, awsEnvVars []corev1.EnvVar) ([]corev1.EnvVar, error) {
+	// Check for secret name
+	secretName, secretKey := r.getSecretNameAndKey(bsl.Spec.Credential, oadpv1alpha1.DefaultPluginAWS)
+
+	r.Log.Info(fmt.Sprintf("Secret name: %s and secret key: %s", secretName, secretKey))
+	// fetch secret and error
+	secret, err := r.getProviderSecret(secretName)
+	if err != nil {
+		r.Log.Info(fmt.Sprintf("Error fetching provider secret %s for backupstoragelocation %s/%s", secretName, bsl.Namespace, bsl.Name))
+		return nil, err
+	}
+
+	r.Log.Info(fmt.Sprintf("Fetched secret: %#v", secret))
+	// parse the secret and get aws access_key and aws secret_key
+	AWSAccessKey, AWSSecretKey, err := r.parseAWSSecret(secret, secretKey)
+	if err != nil {
+		r.Log.Info(fmt.Sprintf("Error parsing provider secret %s for backupstoragelocation %s/%s", secretName, bsl.Namespace, bsl.Name))
+		return nil, err
+	}
+
+	r.Log.Info(fmt.Sprintf("AWSAccessKey: %s, AWSSecretKey: %s", AWSAccessKey, AWSSecretKey))
```

Comment:
We don't want to print the credentials I don't think. Technically they are sensitive

Reply:
Thanks @dymurray , this was an unintentional error on my end, (those were my debug logs :) )
Removed all the inappropriate logs.